pkg/cluster/cluster.go

```diff
@@ -47,9 +47,14 @@ func (b *ClusterBuilder) ResourceBuilders() ([]resource.Builder, error) {
 		if err != nil {
 			return nil, err
 		}
+
 		builders = append(builders, resource.NewServiceAccountBuilder(serviceName, b.Instance, b.Scheme, specs))
 		builders = append(builders, resource.NewDeploymentBuilder(serviceName, b.Instance, b.Scheme, specs))
-		builders = append(builders, resource.NewHeadlessServiceBuilder(serviceName, b.Instance, b.Scheme, specs))
+
+		if serviceName != "worker" {
```

Comment:
Nice catch! Could you please use the temporal constant ? 
```suggestion
		if serviceName != common.WorkerServiceName {
```